src/main/java/org/fenixedu/academic/domain/candidacy/IngressionType.java

```diff
@@ -0,0 +1,271 @@
+package org.fenixedu.academic.domain.candidacy;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.commons.i18n.I18N;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+import pt.ist.fenixframework.Atomic;
+import pt.ist.fenixframework.Atomic.TxMode;
+
+public class IngressionType extends IngressionType_Base {
+
+    private IngressionType() {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    @Atomic(mode = TxMode.WRITE)
+    public static IngressionType createIngressionType(String code, LocalizedString fullDescription) {
+        IngressionType ingressionType = new IngressionType();
+        ingressionType.setCode(code);
+        ingressionType.setFullDescription(fullDescription);
+        return ingressionType;
+    }
+
+    public String getDescription() {
```

Comment:
The preferred pattern is to expose a single getter/setter method for the description returning a LocalizedString, instead of having a getFullDescription(), a getDescription() and getDescription(Locale).

Places where you would use this getDescription() just use <thatGetter>.getContent(), and places where you would use getDescription(Locale) just use <thatGetter>.getContent(Locale). Simpler and more readable.

In the JSPs just go with <thatGetter>.content